assets/jsons/generosity_challenge/chat_scripts/chat_script_day_0.json

```diff
@@ -11,7 +11,7 @@
       "next": {
         "type": "textMessage",
         "side": "interlocutor",
-        "text": "Great! Hi Peterson Family! I believe you're exactly what Tulsa needs to turn things around",
+        "text": "Great! Hi lastName Family! I believe you're exactly what Tulsa needs to turn things around",
```

Comment:
1) There's currently a risk that one of the keys that you are using will now be unintenionally replace text that does not need/want replacement. For example one of your keys is "email". Therefore a text like "We will send you an email at..." or "You might receive some emails..." will actually get replaced.

2) Having the enum ChatScriptSaveKey actually makes the use of the json a bit inflexible. It means that for every key we need to save we would have to implement it on the front-end. Ideally we could just change the hosted json files and not need to care about any app changes anymore.

Suggestion: Since you store and retrieve your user data as a Map<String, dynamic> I think both issues could be fixed by making use of a "Generic Template String". 

https://stackoverflow.com/questions/57230201/generic-template-string-like-in-python-in-dart/57231174#57231174
https://gist.github.com/creativecreatorormaybenot/7c57666a53cbf21f2adab08b4ad84ef5

In the json, the part you want to replace, for example "lastName" would become "{lastName}". You can just delete the ChatScriptSaveKey class. In _formatChatTextWithUserData you would then do:

return TemplateString(source).format(userData) instead of the for loop you have now.

Reply:
@TammiLion , thank you for the review!
Good point about the key words in texts, thanks. They are with brackets {} now.
Re TemplateString - I’ve found a similar https://pub.dev/packages/format package and the formatting part definitely looks more compact now.
I’m not sure about getting rid of ChatScriptSaveKey enum, since it was the point to have specific list of values we gonna use for both user registration and family profiles creation.
Amplitude tracking for the chat is also fixed (kids-962)

Reply:
Understood!